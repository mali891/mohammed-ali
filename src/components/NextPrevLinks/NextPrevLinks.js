import React from 'react'
import PropTypes from 'prop-types'

import { ContentGroup, Link } from '../'
import { Arrow } from '../../icons'
import { styles } from './NextPrevLinks.styles'

const NextPrevLinks = ({ next, prev }) => {
  return (
    <ContentGroup css={styles} className="c-next-prev-links">
      {prev && (
        <p className={`c-next-prev-links__prev ${!next ? 'c-next-prev-links--only' : ''}`}>
          <Arrow className="c-next-prev-links__arrow c-next-prev-links__arrow--prev" />
          Previous post
          <br />
          <Link to={prev.frontmatter.path}>{prev.frontmatter.title}</Link>
        </p>
      )}
      {next && (
        <p className={`c-next-prev-links__next ${!prev ? 'c-next-prev-links--only' : ''}`}>
          Next post
          <Arrow className="c-next-prev-links__arrow c-next-prev-links__arrow--next" />
          <br />
          <Link to={next.frontmatter.path}>{next.frontmatter.title}</Link>
        </p>
      )}
    </ContentGroup>
  )
}

NextPrevLinks.propTypes = {
  next: PropTypes.object,
  prev: PropTypes.object
}

export default NextPrevLinks
