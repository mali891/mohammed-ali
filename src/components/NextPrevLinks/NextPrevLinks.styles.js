import { css } from '@emotion/react'
import { SPACING, MEDIA_QUERIES } from '../../styles/settings'

export const styles = css`
  display: block;
  text-align: center;

  .c-next-prev-links__prev {
    margin-bottom: ${SPACING.MOBILE.SM};
  }

  .c-next-prev-links__arrow {
    width: 1.5rem;

    &--prev {
      transform: rotate(180deg);
      margin-right: 1rem;
    }

    &--next {
      margin-left: 1rem;
    }
  }

  @media (min-width: ${MEDIA_QUERIES.NOTEBOOK}) {
    display: flex;
    align-items: flex-start;
    justify-content: space-between;
    text-align: left;

    .c-next-prev-links__prev,
    .c-next-prev-links__next {
      max-width: 50%;
    }

    .c-next-prev-links__prev {
      margin-bottom: unset;
      padding-right: 1rem;
    }

    .c-next-prev-links__next {
      text-align: right;
      padding-left: 1rem;

      &.c-next-prev-links--only {
        margin-left: auto;
      }
    }

    .c-next-prev-links__arrow {
      width: 2rem;
    }
  }
`
