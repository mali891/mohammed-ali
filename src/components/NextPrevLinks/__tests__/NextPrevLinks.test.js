import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import NextPrevLinks from '../NextPrevLinks'

const defaultProps = {
  next: {
    frontmatter: {
      path: '/blog/roll-your-own-react-tabs',
      title: 'Roll Your Own: React Tabs',
      tags: ['eS6', 'javascript', 'react']
    }
  },
  prev: {
    frontmatter: {
      path: '/blog/roll-your-own-scss-grid',
      title: 'Roll Your Own: SCSS Grid',
      tags: ['eS6', 'javascript', 'react']
    }
  }
}

describe('NextPrevLinks', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<NextPrevLinks {...defaultProps} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders correctly without a next link prop', () => {
    const { asFragment, container } = render(<NextPrevLinks {...defaultProps} next={null} />)

    expect(container.querySelector('.c-next-prev-links__arrow--prev')).toBeInTheDocument()
    expect(container.querySelector('.c-next-prev-links__arrow--next')).not.toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders correctly without a prev link prop', () => {
    const { asFragment, container } = render(<NextPrevLinks {...defaultProps} prev={null} />)

    expect(container.querySelector('.c-next-prev-links__arrow--next')).toBeInTheDocument()
    expect(container.querySelector('.c-next-prev-links__arrow--prev')).not.toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
})
