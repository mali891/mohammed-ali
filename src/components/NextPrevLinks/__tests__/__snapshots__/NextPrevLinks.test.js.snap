// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextPrevLinks renders correctly 1`] = `
<DocumentFragment>
  <section
    class="c-next-prev-links c-content-group"
    css="[object Object]"
  >
    <p
      class="c-next-prev-links__prev "
    >
      <svg
        class="u-fill--secondary u-stroke--secondary c-next-prev-links__arrow c-next-prev-links__arrow--prev"
        viewBox="0 0 477.175 477.175"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M360.731 229.075l-225.1-225.1c-5.3-5.3-13.8-5.3-19.1 0s-5.3 13.8 0 19.1l215.5 215.5-215.5 215.5c-5.3 5.3-5.3 13.8 0 19.1 2.6 2.6 6.1 4 9.5 4 3.4 0 6.9-1.3 9.5-4l225.1-225.1c5.3-5.2 5.3-13.8.1-19z"
        />
      </svg>
      Previous post
      <br />
      <a
        class="c-link u-clr--secondary"
        css="[object Object]"
        href="/blog/roll-your-own-scss-grid"
      >
        Roll Your Own: SCSS Grid
      </a>
    </p>
    <p
      class="c-next-prev-links__next "
    >
      Next post
      <svg
        class="u-fill--secondary u-stroke--secondary c-next-prev-links__arrow c-next-prev-links__arrow--next"
        viewBox="0 0 477.175 477.175"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M360.731 229.075l-225.1-225.1c-5.3-5.3-13.8-5.3-19.1 0s-5.3 13.8 0 19.1l215.5 215.5-215.5 215.5c-5.3 5.3-5.3 13.8 0 19.1 2.6 2.6 6.1 4 9.5 4 3.4 0 6.9-1.3 9.5-4l225.1-225.1c5.3-5.2 5.3-13.8.1-19z"
        />
      </svg>
      <br />
      <a
        class="c-link u-clr--secondary"
        css="[object Object]"
        href="/blog/roll-your-own-react-tabs"
      >
        Roll Your Own: React Tabs
      </a>
    </p>
  </section>
</DocumentFragment>
`;

exports[`NextPrevLinks renders correctly without a next link prop 1`] = `
<DocumentFragment>
  <section
    class="c-next-prev-links c-content-group"
    css="[object Object]"
  >
    <p
      class="c-next-prev-links__prev c-next-prev-links--only"
    >
      <svg
        class="u-fill--secondary u-stroke--secondary c-next-prev-links__arrow c-next-prev-links__arrow--prev"
        viewBox="0 0 477.175 477.175"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M360.731 229.075l-225.1-225.1c-5.3-5.3-13.8-5.3-19.1 0s-5.3 13.8 0 19.1l215.5 215.5-215.5 215.5c-5.3 5.3-5.3 13.8 0 19.1 2.6 2.6 6.1 4 9.5 4 3.4 0 6.9-1.3 9.5-4l225.1-225.1c5.3-5.2 5.3-13.8.1-19z"
        />
      </svg>
      Previous post
      <br />
      <a
        class="c-link u-clr--secondary"
        css="[object Object]"
        href="/blog/roll-your-own-scss-grid"
      >
        Roll Your Own: SCSS Grid
      </a>
    </p>
  </section>
</DocumentFragment>
`;

exports[`NextPrevLinks renders correctly without a prev link prop 1`] = `
<DocumentFragment>
  <section
    class="c-next-prev-links c-content-group"
    css="[object Object]"
  >
    <p
      class="c-next-prev-links__next c-next-prev-links--only"
    >
      Next post
      <svg
        class="u-fill--secondary u-stroke--secondary c-next-prev-links__arrow c-next-prev-links__arrow--next"
        viewBox="0 0 477.175 477.175"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M360.731 229.075l-225.1-225.1c-5.3-5.3-13.8-5.3-19.1 0s-5.3 13.8 0 19.1l215.5 215.5-215.5 215.5c-5.3 5.3-5.3 13.8 0 19.1 2.6 2.6 6.1 4 9.5 4 3.4 0 6.9-1.3 9.5-4l225.1-225.1c5.3-5.2 5.3-13.8.1-19z"
        />
      </svg>
      <br />
      <a
        class="c-link u-clr--secondary"
        css="[object Object]"
        href="/blog/roll-your-own-react-tabs"
      >
        Roll Your Own: React Tabs
      </a>
    </p>
  </section>
</DocumentFragment>
`;
